src/mod_muc_db_rdbms.erl

```diff
@@ -0,0 +1,255 @@
+-module(mod_muc_db_rdbms).
+-include("mod_muc.hrl").
+-include("mongoose_logger.hrl").
+
+-export([init/2,
+         store_room/4,
+         restore_room/3,
+         forget_room/3,
+         get_rooms/2,
+         can_use_nick/4,
+         get_nick/3,
+         set_nick/4,
+         unset_nick/3
+        ]).
+
+-define(ESC(T), mongoose_rdbms:use_escaped_string(mongoose_rdbms:escape_string(T))).
+
+%% Defines which RDBMS pool to use
+%% Parent host of the MUC service
+-type server_host() :: ejabberd:server().
+
+%% Host of MUC service
+-type muc_host() :: ejabberd:server().
+
+%% User's JID. Can be on another domain accessable over FED.
+%% Only bare part (user@host) is important.
+-type client_jid() :: ejabberd:jid().
```

Comment:
There is no such type as `ejabberd:jid()` I wonder if we can catch such mistakes with `dialyzer` or other tools. Do we mean the following?
```suggestion
-type client_jid() :: jid:jid().
```